### PR TITLE
Add a recipe for globus-sdk

### PR DESCRIPTION
**Pull request by xylar**
Add a recipe for the Globus SDK for Python

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge/help-python, I have 2 recipes for related packages (Globus SDK and Globus CLI).  My plan is to submit them as separate PRs but wanted to check what the recommended approach is.

**Comment by xylar** [comment]
@jhkennedy, it looks like I'm already running into trouble.  Any idea why circle CI is failing so quickly?

**Comment by xylar** [comment]
@conda-forge/help-python, I could use some assistance understanding why circleci is failing:
```
#!/bin/bash -eo pipefail
./scripts/run_docker_build.sh
/bin/bash: ./scripts/run_docker_build.sh: No such file or directory
Exited with code 1
```
Seems like I must have done something seriously wrong but I'm not sure what.

**Comment by chrisburr** [comment]
Can you try rebasing on top of conda-forge's master branch?

File: recipes/globus-sdk/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/globus/globus-sdk-python/archive/{{ version }}.tar.gz
+  sha256: 202dadb9e011d3846d71638e79717c192bc431ce8e7fc8b900adbfd4e4dd9407
+
+build:
+  number: 0
+  skip: true  # [win]
```
Comment (chrisburr) [inline recipes/globus-sdk/meta.yaml]: Could this be a `noarch: python` package instead of skipping Windows?

File: recipes/globus-sdk/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  summary: Globus SDK for Python
+  description: This SDK provides a convenient Pythonic interface to Globus APIs.
+  doc_url: http://globus-sdk-python.readthedocs.io/
+  dev_url: https://github.com/globus/globus-sdk-python
+
+extra:
+  recipe-maintainers:
+    - xylar
+    - lukaszlacinski
+    - rpwagner
```
Comment (chrisburr) [inline recipes/globus-sdk/meta.yaml]: @lukaszlacinski and @rpwagner Are you happy to be maintainers of this recipe?

File: recipes/globus-sdk/meta.yaml (hunk 1 above)
Comment (xylar) [reply recipes/globus-sdk/meta.yaml]: I will try that again.  The Windows build failed but perhaps that's not critical, depending on the reason for it.

Comment (chrisburr) [reply recipes/globus-sdk/meta.yaml]: If it was the Azure win-64 build then that one is an issue with the CI and not the recipe. Provided Appveyor passes it's fine.

**Comment by xylar** [comment]
@chrisburr, thanks so much for your help.  I rebased so we'll see how it goes.

File: recipes/globus-sdk/meta.yaml (hunk 1 above)
Comment (xylar) [reply recipes/globus-sdk/meta.yaml]: Okay, that's very good to know.  The Azure builds are pretty new (at least to me) so I wasn't quite sure what to make of the failure.  I've also never tried building this package for Windows myself so had no idea if it was expected to work or not.

**Comment by chrisburr** [comment]
No problem! Assuming Travis passes (which I expect it will, it's just backlogged) I think this is ready to merge once the other two maintainers confirm.

**Comment by xylar** [comment]
Thanks again @chrisburr!  I sent @lukaszlacinski and @rpwagner a slack message so hopefully they'll respond soon.  If not, I can take them off for now and add them back later.

File: recipes/globus-sdk/meta.yaml (hunk 2 above)
Comment (lukaszlacinski) [reply recipes/globus-sdk/meta.yaml]: @chrisburr I am.

Comment (xylar) [reply recipes/globus-sdk/meta.yaml]: Okay, I'll do that.

**Comment by xylar** [comment]
I really appreciate it, @chrisburr!

**Pull request merged**